File: core/src/main/java/com/mcreater/amclcore/model/oauth/session/MinecraftNameChangeableRequestModel.java
package com.mcreater.amclcore.model.oauth.session;

import com.mcreater.amclcore.annotations.RequestModel;
import lombok.Builder;
import lombok.Data;

@Data
@Builder
@RequestModel
public class MinecraftNameChangeableRequestModel {
    private State status;

    public enum State {
        DUPLICATE,
        AVAILABLE,
        NOT_ALLOWED;

        public static State parse(String s) {
            try {
                return valueOf(s);
            } catch (Exception e) {
                return DUPLICATE;
            }
        }
    }
}

File: core/src/main/java/com/mcreater/amclcore/model/oauth/session/MinecraftNameChangeableRequestModel.kt
package com.mcreater.amclcore.model.oauth.session

import com.mcreater.amclcore.annotations.RequestModel

@RequestModel
data class MinecraftNameChangeableRequestModel(
    var status: State? = null
) {
    enum class State {
        DUPLICATE,
        AVAILABLE,
        NOT_ALLOWED;

        companion object {
            @JvmStatic
            fun parse(s: String?): State {
                return try {
                    valueOf(s!!)
                } catch (e: Exception) {
                    DUPLICATE
                }
            }
        }
    }

}
